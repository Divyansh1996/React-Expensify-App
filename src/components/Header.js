import {Link, NavLink} from "react-router-dom";
import React from "react";
import { connect } from "react-redux";
import { startLogout } from "../actions/auth";

export const Header = ({startLogout}) => (
    <div>
        <h1>Expensify App</h1>
        {/* <Link to = "/">DashboardPage Link</Link><br/>
        <Link to = "/create">Create Page Link</Link><br/>
        <Link to = "/edit">Edit Page Link</Link><br/>
        <Link to = "/help">Help Page Link</Link><br/> */}
        <NavLink to = "/dashboard" activeClassName="nav-properties" exact={true}>DashboardPage Link</NavLink><br/>
        <NavLink to = "/create" activeClassName="nav-properties">Create Page Link</NavLink><br/>
        <NavLink to = "/edit" activeClassName="nav-properties">Edit Page Link</NavLink><br/>
        <NavLink to = "/help" activeClassName="nav-properties">Help Page Link</NavLink><br/>
        <button onClick={startLogout}>Logout</button>
    </div>
)

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);