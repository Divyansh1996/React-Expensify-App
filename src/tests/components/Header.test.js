import React from "react";
import { shallow } from "enzyme";
import Header from "../../components/Header"

test("Doing the testing for Header", () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()
})