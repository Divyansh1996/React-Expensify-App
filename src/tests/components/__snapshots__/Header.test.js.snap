// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Doing the testing for Header 1`] = `
<div>
  <h1>
    Expensify App
  </h1>
  <NavLink
    activeClassName="nav-properties"
    exact={true}
    to="/"
  >
    DashboardPage Link
  </NavLink>
  <br />
  <NavLink
    activeClassName="nav-properties"
    to="/create"
  >
    Create Page Link
  </NavLink>
  <br />
  <NavLink
    activeClassName="nav-properties"
    to="/edit"
  >
    Edit Page Link
  </NavLink>
  <br />
  <NavLink
    activeClassName="nav-properties"
    to="/help"
  >
    Help Page Link
  </NavLink>
  <br />
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
